### ISSUE TITLE
Run Checks in Parallel

### ISSUE THREAD

**Issue by shaobo-he** [smackers/smack#750]
It seems to verify a bunch of assertions at the same time may be too difficult for Z3. For example, when we enable memory safety checking or integer overflow checking, we have assertions in almost every function. I think maybe it's a good idea to run the back-end verifier at function- or assertion-granularity in parallel and see if we can boost the performance of SMACK.

**Comment by shaobo-he** [comment]
Merged into #751, which now tracks both pruning trivially proven checks and splitting/running remaining backend checks independently or in parallel.
